developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,247 @@
+                                1, 0, 0, 0, 416, 1321600, 192, 256, 0, 0, 0,
+                                2, 0, 0, 0, 416, 1786000, 192, 256, 0, 0, 0,
+                                3, 0, 0, 0, 416, 2333000, 48, 64, 0, 0, 0,
+                                4, 0, 0, 0, 416, 2910000, 192, 256, 0, 0, 0,
+                                5, 0, 0, 0, 416, 3441000, 192, 256, 0, 0, 0,
+                                6, 0, 0, 0, 416, 4265000, 192, 256, 0, 0, 0]
+
+Module entry definition
+***********************
+| **name**: name of the module
+| **uuid**: uuid of the module
+| **affinity_mask**: usable dsp core mask, bit N is for core N, 0x3 means core 0 and 1 can be used
+| **instance_count**: max module instance count can be active in fw
+| **domain_types**: two schedule domains: domain ll (0), and dp (1)
+| **load_type**: build-in module (BIT(0)), loadable (BIT(1)) or both BIT(0) | BIT(1)
+| **module_type**: defined in the following enum module_type
+
+.. code-block:: bash
+
+  enum module_type {
+        ebasefw = 0,
+        emixin,
+        emixout,
+        ecopier,
+        epeakvol,
+        eupdwmix,
+        emux,
+        esrc,
+        ewov,
+        efx,
+        eaec,
+        ekpb,
+        emicselect,
+        efxf,   /*i.e.SmartAmp */
+        eaudclass,
+        efakecopier,
+        eiodriver,
+        ewhm,
+        egdbstub,
+        esensing,
+        emax,
+        einvalid = emax
+   };
+
+| **auto_start**: Indicates whether a module instance of the module should be created at the base fw startup
+| **sched_caps**: It is a deprecated data used by driver to calculate schedule info for module. Please use the
+| default frame_length 1 and all multiple supported.
+
+.. code-block:: bash
+
+    struct SchedulingCapabilities {
+        uint16_t   frame_length; // Input frame length in sample groups
+        Multiples  multiples_supported;
+    };
+
+    struct Multiples {
+        uint16_t x1 : 1;   // (base * 1)
+        uint16_t x2 : 1;   // (base * 2)
+        uint16_t x3 : 1;   // (base * 3)
+        uint16_t x4 : 1;   // (base * 4)
+        uint16_t x5 : 1;   // (base * 5)
+        uint16_t x6 : 1;   // (base * 6)
+        uint16_t x7 : 1;   // (base * 7)
+        uint16_t x8 : 1;   // (base * 8)
+        uint16_t x9 : 1;   // (base * 9)
+        uint16_t x10 : 1;  // (base * 10)
+        uint16_t x11 : 1;  // (base * 11)
+        uint16_t x12 : 1;  // (base * 12)
+        uint16_t x13 : 1;  // (base * 13)
+        uint16_t x14 : 1;  // (base * 14)
+        uint16_t x15 : 1;  // (base * 15)
+        uint16_t all : 1;  // (base * X)
+    };
+
+| **pin**: is array of data used to define the capability of input & output
+
+.. code-block:: bash
+
+  pin = [dir, type, sample rates, sample size, sample container size, channel config]
+  For example, gain module pin = [0, 0, 0xfeef, 0xf, 0xf, 0x45ff], it supports input pcm
+  supported sample rate [0xxfeef]: 8000, 11500, 16000, 1200, 22050, 24000, 32000, 44100,
```

Comment:
1200 is certainly a typo
I don't known what 11500 is. Should it be 11025?

Reply:
sure, it is 12000 and should be before 16000.